forc-plugins/forc-node/src/run_opts.rs

```diff
@@ -0,0 +1,155 @@
+pub struct RunOpts {
+    pub(crate) service_name: Option<String>,
+    /// DB type, possible options are: `["in-memory", "rocksdb"]`.
+    pub(crate) db_type: DbType,
+    /// Should be used for local development only. Enabling debug mode:
+    /// - Allows GraphQL Endpoints to arbitrarily advance blocks.
+    /// - Enables debugger GraphQL Endpoints.
+    /// - Allows setting `utxo_validation` to `false`.
+    pub(crate) debug: bool,
+    /// Snapshot from which to do (re)genesis.
+    pub(crate) snapshot: PathBuf,
+    /// Peering private key from generated key-pair.
+    pub(crate) keypair: Option<String>,
+    /// Ethereum RPC endpoint.
+    pub(crate) relayer: Option<String>,
+    /// The IP address to bind the GraphQL service to.
+    pub(crate) ip: Option<std::net::IpAddr>,
+    /// The port to bind the GraphQL service to.
+    pub(crate) port: Option<u16>,
+    /// p2p network's TCP port.
+    pub(crate) peering_port: Option<u16>,
+    /// The path to the database, only releavant if the db type is not
+    /// "in-memory".
+    pub(crate) db_path: Option<PathBuf>,
+    /// Enable full utxo stateful validation.
+    pub(crate) utxo_validation: bool,
+    /// Use instant block production mode.
+    /// Newly submitted txs will immediately trigger the production of the next block.
+    pub(crate) poa_instant: bool,
+    /// Enable P2P. By default, P2P is disabled.
+    pub(crate) enable_p2p: bool,
+    /// Addresses of the bootstrap nodes
+    /// They should contain PeerId within their `Multiaddr`.
+    pub(crate) bootstrap_nodes: Option<String>,
+    /// The maximum number of headers to request in a single batch.
+    pub(crate) sync_header_batch_size: Option<u32>,
+    /// Enable the Relayer. By default, the Relayer is disabled.
+    pub(crate) enable_relayer: bool,
+    /// Ethereum contract address for the relayer. Requires conversion of EthAddress into fuel_types.
+    pub(crate) relayer_listener: Option<String>,
+    /// Number of da block that the contract is deployed at.
+    pub(crate) relayer_da_deploy_height: Option<u32>,
+    /// Number of pages or blocks containing logs that
+    /// should be downloaded in a single call to the da layer
+    pub(crate) relayer_log_page_size: Option<u32>,
+    /// The maximum number of get transaction requests to make in a single batch.
+    pub(crate) sync_block_stream_buffer_size: Option<u32>,
+}
+
+#[derive(Debug)]
+pub enum DbType {
+    InMemory,
+    RocksDb,
+}
+
+impl Default for DbType {
+    /// By default fuel-core interprets lack of explicit db-type declaration as
+    /// db-type = rocks-db.
+    fn default() -> Self {
+        Self::RocksDb
+    }
+}
+
+impl Display for DbType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            DbType::InMemory => write!(f, "in-memory"),
+            DbType::RocksDb => write!(f, "rocks-db"),
+        }
+    }
+}
+
+impl RunOpts {
+    pub fn generate_params(self) -> Vec<String> {
```

Comment:
nit: Could impl trait; so can `into()` since `Vec` is a common type.

```suggestion
impl From<RunOpts> for Vec<String> {
    fn from(opts: RunOpts) -> Self {
```

Reply:
This is not a canonical transformation. One would not expect this to be converted to vec string as params out of this context so I specifically created an explicit function for it.